DataFormats/PatCandidates/interface/PackedCandidate.h

```diff
@@ -443,20 +397,37 @@ namespace pat {
       if (numberOfPixelHits_ > trackPixelHitsMask) numberOfPixelHits_ = trackPixelHitsMask;
       int numberOfStripHits_ = tk.hitPattern().numberOfValidHits() - numberOfPixelHits_ - numberOfPixelLayers_ - numberOfStripLayers_;
       if (numberOfStripHits_ > trackStripHitsMask) numberOfStripHits_ = trackStripHitsMask;
+
       packedHits_ = (numberOfPixelHits_&trackPixelHitsMask) | (numberOfStripHits_ << trackStripHitsShift);
- 
+    }
+  
+    virtual void setTrackProperties( const reco::Track & tk, const reco::Track::CovarianceMatrix & covariance,int quality,int covarianceVersion) {
+      covarianceVersion_ = covarianceVersion ;
+      covarianceSchema_ = quality ;
+      normalizedChi2_ = tk.normalizedChi2();
+      //
+      setHits(tk); //tk.hitPattern().numberOfValidPixelHits(),tk.hitPattern().numberOfValidHits());
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant